compiler/src/compiler/rcst.rs

```diff
@@ -410,10 +412,15 @@ impl<T: SplitOuterTrailingWhitespace> SplitOuterTrailingWhitespace for Option<T>
 }
 
 impl<A: SplitOuterTrailingWhitespace, B: SplitOuterTrailingWhitespace> SplitOuterTrailingWhitespace
-    for (A, B)
+    for (A, Vec<B>)
 {
     fn split_outer_trailing_whitespace(self) -> (Vec<Rcst>, Self) {
-        let (whitespace, second) = self.1.split_outer_trailing_whitespace();
-        (whitespace, (self.0, second))
+        if self.1.is_empty() {
+            let (whitespace, first) = self.0.split_outer_trailing_whitespace();
+            (whitespace, (first, self.1))
+        } else {
+            let (whitespace, second) = self.1.split_outer_trailing_whitespace();
+            (whitespace, (self.0, second))
+        }
```

Comment:
Maybe something like this is more readable?

```suggestion
        let (left, right) = self;
        if right.is_empty() {
            let (whitespace, first) = left.split_outer_trailing_whitespace();
            (whitespace, (first, right))
        } else {
            let (whitespace, second) = right.split_outer_trailing_whitespace();
            (whitespace, (left, second))
        }
```